internal/store/store.go

```diff
@@ -58,6 +58,8 @@ type BlockStore struct {
 	mtx    cmtsync.RWMutex
 	base   int64
 	height int64
+
+	dbKeyLayout BlockKeyLayout
```

Comment:
Am I wrong in thinking that the `BlockStore` API is at a too low abstraction level to be effectively implemented in all suitable database backends?

It assumes a global key-value store and segregates relations with prefixes, while using multiple separate KV trees (if supported by the DB engine, which exist in the wild) might allow tuning specifically for different logical data tables. In particular, this API rules out an efficient SQL implementation.

This is an architectural musing, not necessarily calling for action in this PR.

Reply:
@mzabaluev that's something to keep in mind, for sure 👍 However, I'm skeptical about switching to an SQL DB (SQLite) in `v1` as it requires planning, execution, and evaluation (benchmarking, testing). Maybe this is something we should examine when picking a single DB storage.